test/requester.test.ts

```diff
@@ -0,0 +1,399 @@
+    let requester: Requester;
+    let tokenSpy: jest.SpyInstance;
+    let consoleErrSpy: jest.SpyInstance;
+    let consoleWarnSpy: jest.SpyInstance;
+    let token = 'token';
+    let host = 'http://localhost:1738';
+    let clientId = '1234';
+
+
+    beforeEach(() => {
+      requester = new Requester({
+        host,
+        clientId,
+        auth: new Auth({} as any),
+        messenger: new Messenger({} as any),
+      });
+      tokenSpy = jest.spyOn(requester['auth'], 'getToken');
+      tokenSpy.mockImplementation(() => token);
+      consoleWarnSpy = jest.spyOn(console, 'warn');
+      consoleWarnSpy.mockImplementation(() => {});
+      consoleErrSpy = jest.spyOn(console, 'error');
+      consoleErrSpy.mockImplementation(() => {});
+    });
+
+    afterEach(() => {
+      requester = undefined as any;
+      tokenSpy.mockRestore();
+      consoleWarnSpy.mockRestore();
+      consoleErrSpy.mockRestore();
+    });
+
+    const jsonResp = 'json';
+    const textResp = 'text';
+    it.each([
+      [true, jsonResp],
+      [false, textResp],
+    ])(
+      'correctly formats a non-wally, listed rpc request when it is json? %s',
+      async (isJSon, result) => {
+        global.fetch = jest.fn(
+          () =>
+            Promise.resolve({
+              ok: true,
+              status: 200,
+              headers: { get: () => (isJSon ? 'application/json' : '') },
+              text: () => Promise.resolve(textResp),
+              json: () => Promise.resolve(jsonResp),
+            }) as any
+        );
+
+        const ret = await requester.request({
+          method: RPCMethodName.GET_BALANCE,
+          params: ['foo', 'bar'],
+        } as any);
+
+        expect(ret).toEqual(result);
+        expect(global.fetch).toHaveBeenCalledWith(
+          `${host}/oauth/wallet/send`,
+          expect.objectContaining({
+            headers: {
+              Authorization: `Bearer ${token}`,
+              'Content-Type': 'application/json',
+            },
+            body: `{"method":"${RPCMethodName.GET_BALANCE}","params":["foo","bar"],"clientId":"${clientId}"}`,
+          })
+        );
+      }
+    );
+
+    it.each([
+      [undefined],
+      [{ ok: false }],
+      [{ ok: true, status: 300 }],
+      [
+        {
+          ok: true,
+          status: 200,
+          headers: { get: () => 'application/json' as any },
+          json: () => {
+            throw new Error();
+          },
+        },
+      ],
+      [
+        {
+          ok: true,
+          status: 200,
+          headers: { get: () => '' as any },
+          text: () => {
+            throw new Error();
+          },
+        },
+      ],
+    ])('logs an error and rejects when response is %o', (resp) => {
+      global.fetch = jest.fn(() => Promise.resolve(resp) as any);
+      expect.assertions(1);
+      return requester.request({ method: RPCMethodName.BLOCK_NUMBER }).then(
+        () => {
+          expect(true).toBe(false);
+        },
```

Comment:
Q: Is this supposed to be called or are you using this such that if it calls into this, it'll error out? I think the `expect.assertions(1)` should also cover that case no?

Reply:
haha - that's a very good point 😆